### ISSUE TITLE
Allow to build with system squirrel

### ISSUE THREAD

**Issue by pacho2** [SuperTux/supertux#2325]
**SuperTux version:** 0.6.3
**System information:** *(details about your kernel, operating system, or distribution)*

When going to fix CVE-2022-30292 we noticed that supertux was needed to be fixed too as it bundled a squirrel copy. Would it be possible to allow it to build against the official squirrel to prevent this issues in the future?

Thanks a lot

**Comment by tobbi** [comment]
SuperTux already builds against the local squirrel copy by default. If you check https://github.com/SuperTux/supertux/blob/master/mk/cmake/SuperTux/ProvideSquirrel.cmake you can see that `USE_SYSTEM_SQUIRREL` defaults to `ON` which means it first tries to find squirrel on the system and if it can't find that, it'll use the fallback.

**Comment by tobbi** [comment]
I updated the internal squirrel submodule shortly after the release of the CVE. Is there anything else that needs to be done?

**Comment by pacho2** [comment]
It is fine then

Thanks a lot for the info!

**Comment by pacho2** [comment]
Do you know where can I find the logs for the detection code? I have installed squirrel and sdl_ttf and the bundled copies are still being built, even if passing -DUSE_SYSTEM_SDL2_TTF=ON -DUSE_SYSTEM_SQUIRREL=ON options
http://0x0.st/o3vR.3-r1:202

I have read the CMakeFiles/CMakeOutput.log file but I see no traces of it

thanks

**Comment by pacho2** [comment]
It seems that this code is not included in supertux-0.6.3 tarball... we would need a new snapshot then I think :/

**Comment by tobbi** [comment]
Yes, I think that makes sense, reincarnation of Max. We should've actually started doing that when this issue was first patched.

Do we just patch this one bug in the new release or do we include other things?

**Comment by Semphriss** [comment]
I'd recommend we put this, the fix for Chinese, and a fix for the texture problem on Steam. (I may forget some, feel free to add your suggestions)

I wouldn't advise just releasing latest master as a patch, given that the tiles are currently broken from the new tiles PR. Plus, there are some quite major changes (multiplayer, Android support, Narre's flip rewrite etc) which might bring their own lot of bugs.

**Comment by swagtoy** [comment]
We have optional parameter support hacked in and other little patches so I don't think this can be done. Also hey pacho! o/
